File: 122. Best Time to Buy and Sell Stock II/122. Best Time to Buy and Sell Stock II.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0, ans = 0;
        while(left < prices.length - 1){
            int right = left + 1;
            while(right < prices.length && prices[right] >= prices[right-1]){
                ++right;
            }
            --right;
            if(prices[right] > prices[left]){
                ans += prices[right] - prices[left];
            }
            left = right + 1;
        }
        return ans;
    }
}

File: 122. Best Time to Buy and Sell Stock II/122. Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        for(auto it = prices.begin()+1; it != prices.end(); ++it){
            if(*(it-1) < *it){
                ans += *it - *(it-1);
            }
        }
        return ans;
    }
};